python/cudf/cudf/core/buffer/spill_manager.py

```diff
@@ -380,18 +380,10 @@ def spill_to_device_limit(self, device_limit: int = None) -> int:
         )
         if limit is None:
             return 0
-        ret = 0
-        while True:
-            unspilled = sum(
-                buf.size for buf in self.buffers() if not buf.is_spilled
-            )
-            if unspilled < limit:
-                break
-            nbytes = self.spill_device_memory(nbytes=limit - unspilled)
-            if nbytes == 0:
-                break  # No more to spill
-            ret += nbytes
-        return ret
+        unspilled = sum(
+            buf.size for buf in self.buffers() if not buf.is_spilled
+        )
+        return self.spill_device_memory(nbytes=unspilled - limit)
```

Comment:
IIUC the old code was providing a negative number resulting in the while loop continuing until nbytes = 0. Is that right? If so, how did that not completely tank any benchmarks (or did we just not run benchmarks/measure statistics on the final version that we merged)?